src/ProphecyTrait.php

```diff
@@ -38,11 +38,6 @@ trait ProphecyTrait
      */
     protected function prophesize(?string $classOrInterface = null): ObjectProphecy
     {
-        if (\is_string($classOrInterface)) {
-            \assert($this instanceof TestCase);
-            $this->recordDoubledType($classOrInterface);
```

Comment:
this seems unrelated to the commit message

Reply:
It's not. `recordDoubledType` does not exist anymore on `TestCase`, this was the only blocker that was making it fatal during execution. It's all handled with events on PHPUnit now, but the events are strictly related to PHPUnit mocks, so I don't know if its sensible to use them; maybe we could tackle that in a separate PR?

Reply:
The commit adding it says `Add test to reach 100% coverage`.

anyway, I suggest that increasing test coverage to 100% should be done in a separate PR so that we already get 100% test coverage with the code using PHPUnit 9.x

Reply:
> The commit adding it says Add test to reach 100% coverage.
Probably my mistake while committing

> anyway, I suggest that increasing test coverage to 100% should be done in a separate PR so that we already get 100% test coverage with the code using PHPUnit 9.x

Done in #42